spark/v3.2/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestRewriteDataFilesProcedure.java

```diff
@@ -251,8 +326,8 @@ private void createPartitionTable() {
   }
 
   private void insertData(int filesCount) {
-    ThreeColumnRecord record1 = new ThreeColumnRecord(1, "foo", "detail1");
-    ThreeColumnRecord record2 = new ThreeColumnRecord(2, "bar", "detail2");
+    ThreeColumnRecord record1 = new ThreeColumnRecord(1, "foo", null);
```

Comment:
I missed why was this changed?

Reply:
while adding a new test cases for all possible filters, I wanted some null data so that my NOT NULL filter will not execute compaction (doesn't select data). C3 was never used in the testcases. So reused it with null data.